src/lib/pubkey/dilithium/dilithium_common/dilithium_polynomials.h

```diff
@@ -0,0 +1,1609 @@
+                                           uint16_t nonce, const DilithiumModeConstants& mode)
+         {
+         for(size_t i = 0; i < v.m_vec.size(); ++i)
+            {
+            Polynomial::fill_poly_uniform_eta(v.m_vec[i], seed, nonce++, mode);
+            }
+         }
+
+      /*************************************************
+      * Name:        polyvec_pointwise_acc_montgomery
+      *
+      * Description: Pointwise multiply vectors of polynomials of length L, multiply
+      *              resulting vector by 2^{-32} and add (accumulate) polynomials
+      *              in it. Input/output vectors are in NTT domain representation.
+      *
+      * Arguments:   - Polynomial &w: output polynomial
+      *              - const Polynomial &u: pointer to first input vector
+      *              - const Polynomial &v: pointer to second input vector
+      **************************************************/
+      static void polyvec_pointwise_acc_montgomery(Polynomial& w,
+            const PolynomialVector& u, const PolynomialVector& v)
+         {
+         BOTAN_ASSERT_NOMSG(u.m_vec.size() == v.m_vec.size());
+         BOTAN_ASSERT_NOMSG(!u.m_vec.empty() && !v.m_vec.empty());
+
+         u.m_vec[0].poly_pointwise_montgomery(w, v.m_vec[0]);
+
+         for(size_t i = 1; i < v.m_vec.size(); ++i)
+            {
+            Polynomial t;
+            u.m_vec[i].poly_pointwise_montgomery(t, v.m_vec[i]);
+            w += t;
+            }
+         }
+
+      /*************************************************
+      * Name:        fill_polyvecs_power2round
+      *
+      * Description: For all coefficients a of polynomials in vector ,
+      *              compute a0, a1 such that a mod^+ Q = a1*2^D + a0
+      *              with -2^{D-1} < a0 <= 2^{D-1}. Assumes coefficients to be
+      *              standard representatives.
+      *
+      * Arguments:   - PolynomialVector& v1: reference to output vector of polynomials with
+      *                              coefficients a1
+      *              - PolynomialVector& v0: reference to output vector of polynomials with
+      *                              coefficients a0
+      *              - const PolynomialVector& v: reference to input vector
+      **************************************************/
+      static void fill_polyvecs_power2round(PolynomialVector& v1, PolynomialVector& v0, const PolynomialVector& v)
+         {
+         BOTAN_ASSERT((v1.m_vec.size() == v0.m_vec.size()) && (v1.m_vec.size() == v.m_vec.size()),
+                      "possible buffer overflow! Wrong PolynomialVector sizes.");
+         for(size_t i = 0; i < v1.m_vec.size(); ++i)
+            {
+            Polynomial::fill_polys_power2round(v1.m_vec[i], v0.m_vec[i], v.m_vec[i]);
+            }
+         }
+
+      static bool unpack_sig(std::array<uint8_t, DilithiumModeConstants::SEEDBYTES>& c, PolynomialVector& z,
```

Comment:
A lot of these functions in this header are small and seem fine to inline but this for example is quite large - maybe move it to `dilithium_polynomials.cpp`?

Reply:
Fair enough